allensdk/brain_observatory/behavior/data_objects/trials/trial.py

```diff
@@ -0,0 +1,423 @@
+from typing import List, Dict
+
+import numpy as np
+
+from allensdk import one
+from allensdk.brain_observatory.behavior.data_files import StimulusFile
+from allensdk.brain_observatory.behavior.data_objects import StimulusTimestamps
+from allensdk.brain_observatory.behavior.data_objects.licks import Licks
+from allensdk.brain_observatory.behavior.data_objects.rewards import Rewards
+
+
+class Trial:
+    def __init__(self, trial: dict, start: float, end: float,
+                 behavior_stimulus_file: StimulusFile,
+                 index: int, monitor_delay: float,
+                 stimulus_timestamps: StimulusTimestamps,
+                 licks: Licks, rewards: Rewards, stimuli: dict):
+        self._trial = trial
+        self._start = start
+        self._end = self._calculate_trial_end(
+            trial_end=end, behavior_stimulus_file=behavior_stimulus_file)
+        self._index = index
+        self._data = self._match_to_sync_timestamps(
+            monitor_delay=monitor_delay,
+            stimulus_timestamps=stimulus_timestamps, licks=licks,
+            rewards=rewards, stimuli=stimuli)
+
+    @property
+    def data(self):
+        return self._data
+
+    def _match_to_sync_timestamps(
+            self, monitor_delay: float,
+            stimulus_timestamps: StimulusTimestamps,
+            licks: Licks, rewards: Rewards,
+            stimuli: dict):
+        event_dict = {
+            (e[0], e[1]): {
+                'timestamp': stimulus_timestamps.value[e[3]],
+                'frame': e[3]} for e in self._trial['events']
+        }
+
+        tr_data = {"trial": self._trial["index"]}
+        lick_frames = licks.value['frame'].values
+        timestamps = stimulus_timestamps.value
+        reward_times = rewards.value['timestamps'].values
+
+        # this block of code is trying to mimic
```

Comment:
It looks like this same hack solution is implemented in `/allensdk/brain_observatory/behavior/trials_processing.py` around line 565. Is there anyway we can get the two methods to call a function to impose this fix.

(and, yes, I am aware that I am the one who wrote that code twice)

Reply:
Sorry for the confusion, but that function you are referencing will be superseded by this one here; the legacy function needs to be removed but because it's being used by a test currently I am leaving that for the next ticket which will cleanup all the changes.